task/buildah/0.1/buildah.yaml

```diff
@@ -207,6 +213,13 @@ spec:
       [ -n "$COMMIT_SHA" ] && LABELS+=("--label" "vcs-ref=$COMMIT_SHA")
       [ -n "$IMAGE_EXPIRES_AFTER" ] && LABELS+=("--label" "quay.expires-after=$IMAGE_EXPIRES_AFTER")
 
+      ENTITLEMENT_PATH="/entitlement"
+      if [ -n "${ENTITLEMENT_SECRET}" ] && [ -d "$ENTITLEMENT_PATH" ]; then
+        cp -r /entitlement /tmp/entitlement
```

Comment:
So this might seem puzzling, but I could not find any other solution so that these kind of warnings weren't shown:
```
time="2024-02-19T16:14:14Z" level=warning msg="pkg/bind: error detaching \"/var/tmp/buildah969021357/mnt/buildah-bind-target-11\": permission denied"
time="2024-02-19T16:14:14Z" level=warning msg="pkg/bind: error removing \"/var/tmp/buildah969021357/mnt/buildah-bind-target-11\": device or resource busy"
time="2024-02-19T16:14:14Z" level=warning msg="pkg/bind: error removing \"/var/tmp/buildah969021357/mnt\": directory not empty"
```

They must be related to the permissions of the volume. I tried changing the permissions of the volume, mounting it to different places and nothing worked expected making a copy of the volume and mounting that copy to the buildah. Any ideas?

Reply:
Never seen that :confused: 

I guess copying the secret is fine, but maybe we could copy with `--preserve=mode` to preserve permissions?

Reply:
Updated to use the preserve option. Interestingly, it works without any warnings. This makes the issue of the buildah warnings even more puzzling...